### PR TITLE
Mako ci integration test

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

This change leverages the mako test suite to test whether ddtrace instrumentation alters the functionality of the Mako framework in unexpected ways. 

## Checklist
- [x] Added to the correct milestone.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Library documentation is updated.
- [x] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: .github/workflows/mako.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        with:
+          path: ddtrace
+      - uses: actions/checkout@v2
+        with:
+          repository: sqlalchemy/mako
+          ref: 1.1.4
+          path: mako
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
```
Comment (brettlangdon) [inline .github/workflows/mako.yml:26]: The version of python doesn't matter, we could do 3.9, but 3.8 is stable enough and perfect for now.

File: .github/workflows/mako.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Install tox
+        run: pip install tox
+      - name: Create tox env
+        run: tox -e py38 --notest
```
Comment (brettlangdon) [inline .github/workflows/mako.yml:30]: The mako `tox.ini` only has 1 environment which is just called `py`

```suggestion
        run: tox -e py --notest
```

File: .github/workflows/mako.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Inject ddtrace
+        run: .tox/py38/bin/pip install ../ddtrace
```
Comment (brettlangdon) [inline .github/workflows/mako.yml:32]:
```suggestion
        run: .tox/py/bin/pip install ../ddtrace
```

File: .github/workflows/mako.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Add pytest configuration for ddtrace
+        run: echo -e "[pytest]\nddtrace-patch-all = 1" > pytest.ini
+      - name: Run tests
+        # test_exception_propagation is broken upstream
+        run: tox -e py38 -- -k 'not test_exception_propagation and not test_memory_consumption'
```
Comment (brettlangdon) [inline .github/workflows/mako.yml:37]:
```suggestion
        run: tox -e py
```

Change to "py" env, and we can get rid of the ignores that were copied from the Flask one.

File: .github/workflows/mako.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      - name: Install tox
+        run: pip install tox
+      - name: Create tox env
+        run: tox -e py --notest
+      - name: Inject ddtrace
+        run: .tox/py/bin/pip install ../ddtrace
+      - name: Add pytest configuration for ddtrace
+        run: echo -e "[pytest]\nddtrace-patch-all = 1" > pytest.ini
+      - name: Run tests
+        run: tox -e py
```
Comment (brettlangdon) [inline .github/workflows/mako.yml:36]: missing newline at the end of the file

Comment (mabdinur) [reply .github/workflows/mako.yml:36]: Is there a linter I should set up?

File: .github/workflows/mako.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
```
Comment (brettlangdon) [inline .github/workflows/mako.yml:37]: no idea why this is still showing missing newline at the end of file. might need to tweak your editor.

Comment (mabdinur) [reply .github/workflows/mako.yml:37]: Is there a formatter or linter I should set up?

Comment (mabdinur) [reply .github/workflows/mako.yml:37]: I'm just using vscode

**Review by brettlangdon — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]
👏  yay!

**Review by Yun-Kim — APPROVED** [review]
😄

**Pull request merged**